rpcs3/Emu/Cell/Modules/cellAudioOut.cpp

```diff
@@ -122,29 +124,55 @@ error_code cellAudioOutGetState(u32 audioOut, u32 deviceIndex, vm::ptr<CellAudio
 		return CELL_AUDIO_OUT_ERROR_ILLEGAL_PARAMETER;
 	}
 
-	*state = {};
 
-	switch (audioOut)
+	const auto num = cellAudioOutGetNumberOfDevice(audioOut);
+
+	if (num < 0)
 	{
-	case CELL_AUDIO_OUT_PRIMARY:
-		state->state = CELL_AUDIO_OUT_OUTPUT_STATE_ENABLED;
-		state->encoder = CELL_AUDIO_OUT_CODING_TYPE_LPCM;
-		state->downMixer = CELL_AUDIO_OUT_DOWNMIXER_NONE;
-		state->soundMode.type = CELL_AUDIO_OUT_CODING_TYPE_LPCM;
-		state->soundMode.channel = CELL_AUDIO_OUT_CHNUM_8;
-		state->soundMode.fs = CELL_AUDIO_OUT_FS_48KHZ;
-		state->soundMode.reserved = 0;
-		state->soundMode.layout = CELL_AUDIO_OUT_SPEAKER_LAYOUT_8CH_LREClrxy;
+		return num;
+	}
 
-		return CELL_OK;
+	CellAudioOutState _state;
+	std::memset(&_state, 0, sizeof(_state));
 
+	if (deviceIndex >= num + 0u)
+	{
+		if (audioOut == CELL_AUDIO_OUT_SECONDARY)
+		{
+			// Error codes are not returned here
+			// Random (uninitialized) data from the stack seems to be returned here
+			// Although it was constant on my tests so let's write that
```

Comment:
Doesn't it mean that `state` remains unmodified?

Reply:
No, more like its a copy of another unitiliazed buffer in vsh.
In the hw tests I always memset the variables before calling the functions.

Reply:
Here's the output of it: https://github.com/elad335/myps3tests/blob/master/ppu_tests/cellAudioOut/expected.txt#L178

Reply:
Okay